### PR TITLE
Firewall: Updates detection to warn when falling back to non-compatible xtables driver

### PR DESCRIPTION
**Pull request by tomponline**
Warning contains the issue as to why the xtables driver we are using as "last resort" is not fully compatible and may still cause unexpected behaviour.

### CODE REVIEW

File: lxd/firewall/firewall_load.go [hunk 1]
```diff
@@ -10,26 +11,35 @@ func New() Firewall {
...
+		return nftables
+	}
+
+	// If neither nftables nor xtables are compatible, we fallback to xtables.
+	// This continues the existing behaviour of allowing LXD to start with potentially an incomplete firewall
+	// backend, so that only networks and instances using those features may fail to function properly.
+	// The most common scenario for this is when xtables is using nft shim commands but the nft command itself
+	// is not installed. In this case LXD will use the xtables shim commands but with the potential of problems
+	// due to differences between the original xtables commands and the shim commands provided by nft.
+	logger.Warnf(`Firewall failed to detect any compatible driver, falling back to "xtables" (but some features may not work as expected due to: %v)`, xtablesCompatErr)
```
Comment (stgraber) [inline lxd/firewall/firewall_load.go:43]: Did I miss something or will we be displaying this warning on systems where:
 - nftables isn't available
 - xtables is functional but not in use

Comment (stgraber) [reply lxd/firewall/firewall_load.go:43]: I believe the warning should only be displayed with both nftablesCompatErr and xtablesCompatErr are non-nil

Comment (tomponline) [reply lxd/firewall/firewall_load.go:43]: @stgraber yep you're right, good spot, it was like that original, but I removed it last at minute for some reason.

Comment (tomponline) [reply lxd/firewall/firewall_load.go]: @stgraber fixed.

**Pull request merged**